### PR TITLE
implement search function

### PR DESCRIPTION
**Pull request by phthien241**

### CODE REVIEW

File: career-link-webapi/Controllers/PostController.cs [hunk 1]
```diff
@@ -13,16 +15,80 @@ public PostController(IPostService taskService)
             _postService = taskService;
         }
 
-        [HttpGet("allpost")]
```
Comment (Ben0189) [inline career-link-webapi/Controllers/PostController.cs:16]: if you change the end point path, do you make sure the frontend using this endpoint still working ?

File: career-link-webapi/Controllers/PostController.cs [hunk 2]
```diff
@@ -13,16 +15,80 @@ public PostController(IPostService taskService)
...
-        
+
         [HttpGet("{id}")]
         public async Task<IActionResult> GetTaskByID(int id)
         {
             return Ok(await _postService.GetPostById(id));
         }
+
+        [HttpGet("search")]
+        public async Task<ActionResult<List<PostDTO>>> SearchPost([FromQuery] int experienceLevel, [FromQuery] List<int> skillIds)
```
Comment (Ben0189) [inline career-link-webapi/Controllers/PostController.cs:31]: this endpoint is getting list of skillIds??

File: career-link-webapi/Migrations/20241001123901_SeedDataForPost.cs
Comment (Ben0189) [inline career-link-webapi/Migrations/20241001123901_SeedDataForPost.cs:1]: why remove migrations file

File: career-link-webapi/Migrations/20241003093247_AddSkillsTable.cs
Comment (Ben0189) [inline career-link-webapi/Migrations/20241003093247_AddSkillsTable.cs:1]: why manually changes migrations file

File: career-link-webapi/Services/PostService.cs [hunk 3]
```diff
@@ -23,19 +27,20 @@ public PostService(CareerLinkDbContext dbContext)
...
                         {
-                            PostId = x.PostId, 
-                            User = new UserDTO 
+                            PostId = x.PostId,
+                            User = new UserDTO
                             {
                                 UserId = x.User.Id,
-                                UserName = x.User.UserName,
-                                Email = x.User.Email,
+                                UserName = x.User.UserName ?? "Unknown",
```
Comment (Ben0189) [inline career-link-webapi/Services/PostService.cs:37]: user name is the require field for the post

File: career-link-webapi/Services/PostService.cs [hunk 4]
```diff
@@ -49,15 +54,19 @@ public async Task<PostDTO> GetPostById(int id)
...
-                                    {
-                                        UserId = x.User.Id,
-                                        UserName = x.User.UserName,
-                                        Email = x.User.Email,
-                                    },
+                        PostId = x.PostId,
+                        User = new UserDTO
+                        {
+                            UserId = x.User.Id,
+                            UserName = x.User.UserName ?? "Unknown",
```
Comment (Ben0189) [inline career-link-webapi/Services/PostService.cs:61]: same here username should not be nullable

File: career-link-webapi/Services/PostService.cs [hunk 5]
```diff
@@ -70,8 +79,119 @@ public async Task<PostDTO> GetPostById(int id)
...
+                {
+                    UserId = post.User.Id,
+                    UserName = post.User.UserName ?? "Unknown",
+                    Email = post.User.Email ?? "Unknown",
+                    PhoneNumber = post.User.PhoneNumber ?? "Unknown",
+                    CreatedDate = post.User.CreatedDate,
+                    UpdatedDate = post.User.UpdatedDate
+                },
+                SkillNames = post.Skills.Select(s => s.Name).ToList()
+            }).ToList();
```
Comment (Ben0189) [inline career-link-webapi/Services/PostService.cs:116]: please fix the format here properly this is hard to read

File: career-link-webapi/Services/PostService.cs [hunk 6]
```diff
@@ -70,8 +79,119 @@ public async Task<PostDTO> GetPostById(int id)
...
+                    UserName = newPost.User.UserName ?? "Unknown",
+                    Email = newPost.User.Email ?? "Unknown",
+                    PhoneNumber = newPost.User.PhoneNumber ?? "Unknown",
+                },
+                Description = newPost.Description,
+                experienceLevel = newPost.experienceLevel,
+                CreatedDate = newPost.CreatedDate,
+                UpdatedDate = newPost.UpdatedDate,
+                SkillIds = newPost.Skills.Select(s => s.SkillId).ToList()
+            };
```
Comment (Ben0189) [inline career-link-webapi/Services/PostService.cs:157]: you dont need to return the whole dto, just return the name or simply a string message say success create is enough

File: career-link-webapi/Services/PostService.cs [hunk 7]
```diff
@@ -70,8 +79,119 @@ public async Task<PostDTO> GetPostById(int id)
...
+        }
+
+        public async Task<string> UpdatePost(int postId, PostDTO postDto)
+        {
+            var existingPost = await _dbContext.Posts.Include(p => p.Skills).FirstOrDefaultAsync(p => p.PostId == postId);
```
Comment (Ben0189) [inline career-link-webapi/Services/PostService.cs:162]: fix this format as well, see how i do it

**Review by Ben0189 — CHANGES_REQUESTED** [review]
fixes needed

File: career-link-webapi/Controllers/PostController.cs (hunk 2 above)
Comment (phthien241) [reply career-link-webapi/Controllers/PostController.cs:31]: list of skill names

**Review by Ben0189 — APPROVED** [review]

**Pull request merged**